arduino-ide-extension/src/browser/settings.tsx

```diff
@@ -376,9 +378,18 @@ export class SettingsComponent extends React.Component<SettingsComponent.Props,
                     onChange={this.additionalUrlsDidChange} />
                 <i className='fa fa-window-restore theia-button shrink' onClick={this.editAdditionalUrlDidClick} />
             </div>
+
+            <div className='flex-line'>
+                <button className='theia-button shrink' onClick={this.openExtendedSettings}>Extended Settings</button>
+            </div>
         </div>;
     }
 
+    protected openExtendedSettings = (e: React.MouseEvent<HTMLElement>) => {
+      this.props.commands.executeCommand(CommonCommands.OPEN_PREFERENCES.id);
+      this.props.commands.executeCommand('arduino-settings-close');
```

Comment:
Even if I defined `CommonCommands.CLOSE_PREFERENCES` I'm not even able to use it here, as it causes a loop in the dependencies.

If we go the "commands path" we should move these preference commands to another file, but my understanding is we don't do that usually.